Userland/Libraries/LibUnicode/IDNA.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2023, Simon Wanner <simon@skyrising.xyz>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Error.h>
```

Comment:
This will all fail to compile when the ENABLE_UNICODE_DATABASE_DOWNLOAD CMake option is OFF. We can't rely on headers like UnicodeData.h existing in that case. See other .cpp files in LibUnicode for how to handle this.